crates/ruff_python_ast/src/parenthesize.rs

```diff
@@ -0,0 +1,54 @@
+use ruff_python_trivia::{first_non_trivia_token, SimpleTokenKind, SimpleTokenizer};
+use ruff_text_size::TextRange;
+
+use crate::node::AnyNodeRef;
+use crate::Ranged;
+
+/// A wrapper around an expression that may be parenthesized.
+#[derive(Debug)]
+pub struct ParenthesizedExpression<'a> {
+    /// The underlying AST node.
+    expr: AnyNodeRef<'a>,
```

Comment:
Is the expression read anywhere? Would be nice to have the `ExpressionRef` that I started introducing once.

Reply:
No, only the range. Should I add `ExpressionRef`? (`AnyExpressionRef`?)

Reply:
I guess `AnyExpressionRef` is redundant and `ExpressionRef` is more appropriate.

Reply:
Dug up the old PR, I can reintroduce it separately.